### PR TITLE
Add amplicon length filtering

### LINKED ISSUE #449: Implement post-dada2 amplicon length filtering
### Description of feature

Currently ampliseq only allows filtering of input read lengths, but there is no way (that I've found) to specify a target range of lengths for the amplicons post merging. Working with COX1, we can have a strict length expectation, and know that an amplicon with even one base variation from this is more likely to contain errors, and anything substantially shorter or longer is definitely not of interest. 

It seems like this would be straightforward to implement in the R code in `dada2_denoising.nf`, by including the line 
`seqtab <- seqtab[,nchar(colnames(seqtab)) %in% <minampliconlen>:<maxampliconlen>])` after the current lines 47 and 70, however of course I'm not familiar enough with nextflow to know how complex it might be to implement and pass through the `<*ampliconlen>` parameters. 

ETA: we imagine this would likely be useful for non-COX1 amplicons as well, of course with less strict length ranges. 

Thanks!

### PR DESCRIPTION
**Pull request by d4straub**
Closes #449 .
Adds an optional ASV length filter after barrnap and before ASV taxonomic classification.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: docs/output.md [hunk 1]
```diff
@@ -15,8 +15,9 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
     - [FastQC](#fastqc) - Read quality control
     - [Cutadapt](#cutadapt) - Primer trimming
     - [MultiQC](#multiqc) - Aggregate report describing results
-    - [DADA2](#dada2) - Infer Amplicon Sequence Variants (ASVs) and taxonomic classification
-      - [Barrnap](#barrnap) - Predict ribosomal RNA sequences
+    - [Amplicon sequence variant inferrence with DADA2](#Amplicon-sequence-variant-inferrence-with-DADA2) - Infer Amplicon Sequence Variants (ASVs) and taxonomic classification
+      - [Barrnap](#barrnap) - Predict ribosomal RNA sequences and optional filtering
+      - [Length filter](#length-filter) - Optionally, ASV can be filtered by length thresholds
```
Comment (erikrikarddaniel) [inline docs/output.md:20]: Is the indentation off here?

**Review by erikrikarddaniel — APPROVED** [review]
Tiny question regarding indentation, otherwise fine.

File: docs/output.md (hunk 1 above)
Comment (d4straub) [reply docs/output.md:20]: Hm, indentation of `Length filter` is at the same level as `Barrnap` above & `ITSx` below. And those are sub-sections of `Amplicon sequence variant inferrence with DADA2`, that is not correct.
Maybe if I change the header `Amplicon sequence variant inferrence with DADA2` to `Amplicon sequence variant inferrence with DADA2 and subsequent filtering` than having the sub-sections `Barrnap` & `Length filter` & `ITSx` are fine, but the sub-section `assignSH` still doesn't fit.

Comment (d4straub) [reply docs/output.md]: I attempted to fix this below.

**Pull request merged**